macos/FluentUI/Button/Button.swift

```diff
@@ -5,158 +5,254 @@
+		imagePosition: NSControl.ImagePosition = .imageLeading,
+		format: ButtonFormat = ButtonFormat()
+	)
+	{
+		super.init(frame: .zero)
 		isBordered = false
 		wantsLayer = true
 		layer?.contentsScale = window?.backingScaleFactor ?? 1.0
 
-		self.style = style
+		if let cell = cell as? ButtonCell {
+			// The Button properties `contentTintColorDisabled`,
+			// `backgroundColorDisabled`, and `borderColorDisabled` fully
+			// specify the look of the disabled button, and may already include
+			// a 'disabled' system effect.  Don't apply the effect again on top
+			// of that.
+			cell.imageDimsWhenDisabled = false
+		}
 
 		if let title = title {
 			self.title = title
-		}
-
-		if let image = image {
+			if let image = image {
+				self.image = image
+				self.imagePosition = imagePosition
+			} else {
+				self.imagePosition = .noImage
+			}
+		} else if let image = image {
 			self.image = image
+			self.imagePosition = .imageOnly
 		}
-		
-		if style == .primaryFilled {
-			contentTintColor = .white
-		} else {
-			contentTintColor = primaryColor
-		}
+
+		self.format = format
 	}
 
 	open class override var cellClass: AnyClass? {
 		get {
-			return FluentButtonCell.self
+			return ButtonCell.self
 		}
 		set {}
 	}
-	
-	/// Image display in the button
+
+	/// Image to display in the button.
 	override public var image: NSImage? {
 		willSet {
 			guard wantsLayer == true else {
 				preconditionFailure("wantsLayer must be set so that the image is rendered on the layer")
 			}
 		}
 	}
-	
-	/// String display in the button
+
+	/// Title string to display in the button.
 	override public var title: String {
 		willSet {
 			guard wantsLayer == true else {
 				preconditionFailure("wantsLayer must be set so that the title is rendered on the layer")
 			}
 		}
 	}
-	
-	override public func updateTrackingAreas() {
-		super.updateTrackingAreas()
-		
-		// Remove existing trackingArea
-		if let trackingArea = trackingArea {
-			removeTrackingArea(trackingArea)
-			self.trackingArea = nil
-		}
 
-		// Create a new trackingArea
-		let opts: NSTrackingArea.Options = [.mouseEnteredAndExited, .activeAlways]
-		let trackingArea = NSTrackingArea(
-			rect: bounds,
-			options: opts,
-			owner: self,
-			userInfo: nil
-		)
-		addTrackingArea(trackingArea)
-		self.trackingArea = trackingArea
-	}
+	private var mouseDown = false
 
-	open override func mouseEntered(with event: NSEvent) {
-		mouseEntered = true
-		needsDisplay = true
-	}
+	/// While the current Button is pressed, its style is temporarily applied to the linkedPrimary button.
+	/// This emulates an effect seen in the default style of Cocoa buttons, where pressing a secondary
+	/// button takes the accent color highlighting from a nearby primary button. For best results, the
+	/// current button should have the `.secondary` style, the linkedPrimary button should have the
+	/// `.primary` style, and both buttons should have the same accentColor.
+	@objc public var linkedPrimary: Button?
```

Comment:
When we say "Cocoa Button", do we mean CocoaUI? we probably shouldn't mention internal office frameworks in our open source code. Also, I didn't know about this. I'm guessing this is used in like dialogs and alerts?

Reply:
No, I mean [Cocoa](https://en.wikipedia.org/wiki/Cocoa_(API)) as in NSButton.

Reply:
As distinct from, say, Cocoa Touch and SwiftUI.